analytics/pubstack/eventchannel/eventchannel_test.go

```diff
@@ -4,178 +4,144 @@ import (
 
 	eventChannel.buffer([]byte("one"))
-	eventChannel.buffer([]byte("two"))
 
-	assert.NotEqual(t, eventChannel.metrics.eventCount, int64(0))
-	assert.NotEqual(t, eventChannel.metrics.bufferSize, int64(0))
+	assert.NotZero(t, eventChannel.metrics.eventCount)
+	assert.NotZero(t, eventChannel.metrics.bufferSize)
 
 	eventChannel.reset()
 
-	assert.Equal(t, eventChannel.buff.Len(), 0)
-	assert.Equal(t, eventChannel.metrics.eventCount, int64(0))
-	assert.Equal(t, eventChannel.metrics.bufferSize, int64(0))
+	assert.Zero(t, eventChannel.buff.Len())
+	assert.Zero(t, eventChannel.metrics.eventCount)
+	assert.Zero(t, eventChannel.metrics.bufferSize)
 }
 
-func TestEventChannel_flush(t *testing.T) {
-	data := make([]byte, 0)
-	send := newSender(&data)
+func TestEventChannelFlush(t *testing.T) {
+	dataSent := make(chan []byte)
+	send := newSender(dataSent)
+	clockMock := clock.NewMock()
 
-	eventChannel := NewEventChannel(send, maxByteSize, maxEventCount, maxTime)
+	eventChannel := NewEventChannel(send, clockMock, largeBufferSize, largeEventCount, maxTime)
 	defer eventChannel.Close()
 
 	eventChannel.buffer([]byte("one"))
 	eventChannel.buffer([]byte("two"))
 	eventChannel.buffer([]byte("three"))
 	eventChannel.flush()
-	time.Sleep(10 * time.Millisecond)
 
-	assert.Equal(t, string(data), "onetwothree")
+	data, _ := readChanOrTimeout(t, dataSent)
+	assert.Equal(t, "onetwothree", readGz(data))
 }
 
-func TestEventChannel_close(t *testing.T) {
-	data := make([]byte, 0)
-	send := newSender(&data)
+func TestEventChannelClose(t *testing.T) {
+	dataSent := make(chan []byte)
+	send := newSender(dataSent)
+	clockMock := clock.NewMock()
 
-	eventChannel := NewEventChannel(send, 15000, 15000, 2*time.Hour)
+	eventChannel := NewEventChannel(send, clockMock, largeBufferSize, largeEventCount, maxTime)
 
 	eventChannel.buffer([]byte("one"))
 	eventChannel.buffer([]byte("two"))
 	eventChannel.buffer([]byte("three"))
 	eventChannel.Close()
 
-	time.Sleep(10 * time.Millisecond)
-
-	assert.Equal(t, string(data), "onetwothree")
+	data, _ := readChanOrTimeout(t, dataSent)
+	assert.Equal(t, "onetwothree", readGz(data))
 }
 
-func TestEventChannel_Push(t *testing.T) {
-	data := make([]byte, 0)
-	send := newSender(&data)
+func TestEventChannelPush(t *testing.T) {
+	dataSent := make(chan []byte)
+	send := newSender(dataSent)
+	clockMock := clock.NewMock()
 
-	eventChannel := NewEventChannel(send, 15000, 5, 5*time.Millisecond)
+	eventChannel := NewEventChannel(send, clockMock, largeBufferSize, largeEventCount, 1*time.Second)
 	defer eventChannel.Close()
 
-	eventChannel.Push([]byte("one"))
-	eventChannel.Push([]byte("two"))
-	eventChannel.Push([]byte("three"))
-	eventChannel.Push([]byte("four"))
-	eventChannel.Push([]byte("five"))
-	eventChannel.Push([]byte("six"))
-	eventChannel.Push([]byte("seven"))
-
-	time.Sleep(10 * time.Millisecond)
-
-	assert.Equal(t, string(data), "onetwothreefourfivesixseven")
-
-}
-
-func TestEventChannel_OutputFormat(t *testing.T) {
```

Comment:
I removed this test. It was validating the output by gzip'ing the expected response and comparing against the real response, versus the other tests which validate by un-gzip'ing the actual response to compare against the expected response. I do not see the value in having both forms.